contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -33,52 +37,48 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
-    // Note that these values are also used to determine if a given message has been delivered or not.
-    // The first key is the subnet ID, the second key is the message ID, and the value is the reward address
-    // provided by the deliverer of the uniquely identified message.
-    mapping(bytes32 => mapping(uint256 => address))
-        public relayerRewardAddresses;
-
     // Tracks the hash of messages that have been received but whose execution has never succeeded.
     // Enables retrying of failed messages with higher gas limits. Message execution is guaranteed to
-    // succeed at most once.  The first key is the subnet ID, the second key is the message ID, and
+    // succeed at most once. The first key is the blockchain ID, the second key is the message ID, and
     // the value is the hash of the uniquely identified message whose execution failed.
-    mapping(bytes32 => mapping(uint256 => bytes32))
+    mapping(bytes32 sourceChainID => mapping(uint256 messageID => bytes32 messageHash))
         public receivedFailedMessageHashes;
 
-    // Tracks the fee amounts for a given asset able to be redeemed by a given relayer.
-    // The first key is the relayer address, the second key is the ERC20 token contract address,
-    // and the value is the amount of the asset owed to the relayer.
-    mapping(address => mapping(address => uint256)) public relayerRewardAmounts;
-
-    // Teleporter delivers message by invoking functions of the form: funcName(bytes32 originChainID, address originSenderAddress, ...)
-    // where "..." can represent arbitrary additional parameters. Accounting for the 4 byte function selector and two 32 byte parameters,
-    // there is a minimum valid message length of 68 bytes.
-    uint256 public constant REQUIRED_ORIGIN_CHAIN_ID_START_INDEX = 4;
-    uint256 public constant MINIMUM_REQUIRED_CALL_DATA_LENGTH = 68;
+    // Tracks the relayer reward address for each message delivered from a given subnet.
+    // Note that these values are also used to determine if a given message has been delivered or not.
+    // The first key is the blockchain ID, the second key is the message ID, and the value is the reward address
+    // provided by the deliverer of the uniquely identified message.
+    mapping(bytes32 sourceChainID => mapping(uint256 messageID => address relayerRewardAddress))
+        internal _relayerRewardAddresses;
 
-    // The blockchain ID of the chain the contract is deployed on. Initialized lazily when receiveCrossChainMessage() is called,
-    // if the value has not already been set.
-    bytes32 public blockchainID;
+    // Tracks the fee amounts for a given asset able to be redeemed by a given relayer.
+    // The first key is the relayer address, the second key is the fee token contract address,
+    // and the value is the amount of the asset redeemable by the relayer.
+    mapping(address relayerAddress => mapping(address feeTokenContract => uint256 redeemableRewardAmount))
```

Comment:
can we rename this to `relayerRewardAddress` and update comments?

Reply:
updated now